examples/timeseries/event_classification_for_payment_card_fraud_detection.py

```diff
@@ -0,0 +1,430 @@
+"""
+Title: Event classification for payment card fraud detection
+Author: [achoum](https://github.com/achoum/)
+Date created: 2024/02/01
+Last modified: 2024/02/01
+Description: Detection of fraudulent payment card transactions using Temporian and a feed-forward neural network.
+Accelerator: GPU
+"""
+
+"""
+This notebook depends on Keras Core, Temporian, and a few other libraries. You can install them as follow:
+
+```shell
+pip install temporian keras pandas tf-nightly scikit-learn -U
+```
+"""
+
+import keras  # To train the Machine Learning model
+import temporian as tp  # To convert transactions into tabular data
+
+import numpy as np
+import os
+import pandas as pd
+import datetime
+import math
+import tensorflow as tf
+from sklearn.metrics import RocCurveDisplay
+
+"""
+## Introduction
+
+Payment fraud detection is critical for banks, businesses, and consumers. In Europe alone, fraudulent transactions were estimated at [€1.89 billion in 2019](https://www.ecb.europa.eu/pub/pdf/cardfraud/ecb.cardfraudreport202110~cac4c418e8.en.pdf). Worldwide, approximately [3.6%](https://www.cybersource.com/content/dam/documents/campaign/fraud-report/global-fraud-report-2022.pdf) of commerce revenue is lost to fraud. In this notebook, we train and evaluate a model to detect fraudulent transactions using the synthetic dataset attached to the book [Reproducible Machine Learning for Credit Card Fraud Detection](https://fraud-detection-handbook.github.io/fraud-detection-handbook/Foreword.html) by Le Borgne et al.
+
+Fraudulent transactions often cannot be detected by looking at transactions in isolation. Instead, fraudulent transactions are detected by looking at patterns across multiple transactions from the same user, to the same merchant, or with other types of relationships. To express these relationships in a way that is understandable by a machine learning model, and to augment features with feature engineering. We use the [Temporian](https://temporian.readthedocs.io/en/latest) preprocessing library.
+
+We pre-process an event-set dataset into a tabular dataset and use a feed-forward neural network to learn the patterns of fraud and make predictions
```

Comment:
- preprocess can be written without the hyphen (and would make it consistent with "preprocessing library" in the previous line)
- would s/an event-set dataset/a transactional dataset
- missing full stop at the end